### PR TITLE
add azure provider 

### PR DESCRIPTION
**Pull request by aramase**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,10 @@
+FROM alpine:3.10
+
+WORKDIR /bin
+
+RUN apk add --no-cache bash
+ADD ./secrets-store-csi-driver-provider-azure /bin/secrets-store-csi-driver-provider-azure
+RUN chmod a+x /bin/secrets-store-csi-driver-provider-azure
+ADD ./install.sh /bin/install_azure_provider.sh
+
+ENTRYPOINT ["/bin/install_azure_provider.sh"]
```
Comment (ritazh) [inline Dockerfile]: nit: line break

File: examples/nginx-pod-secrets-store-inline-volume-pod-identity.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+                objectName: secret1
+                objectType: secret        # object types: secret, key or cert
+                objectVersion: ""         # [OPTIONAL] object versions, default to latest if empty
+              - |
+                objectName: key1
+                objectType: key
+                objectVersion: ""
+          resourceGroup: ""               # the resource group of the KeyVault
+          subscriptionId: ""              # the subscription ID of the KeyVault
+          tenantId: ""                    # the tenant ID of the KeyVault
```
Comment (ritazh) [inline examples/nginx-pod-secrets-store-inline-volume-pod-identity.yaml]: nit: line break

File: install.sh [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+azure_provider_dir="${target_dir}/azure"
+mkdir -p ${azure_provider_dir}
+
+cp /bin/secrets-store-csi-driver-provider-azure ${azure_provider_dir}/provider-azure
+
+#https://github.com/kubernetes/kubernetes/issues/17182
+# if we are running on kubernetes cluster as a daemon set we should
+# not exit otherwise, container will restart and goes into crashloop (even if exit code is 0)
+while true; do echo "install done, daemonset sleeping" && sleep 60; done
```
Comment (ritazh) [inline install.sh]: nit: line break

**Comment by ritazh** [comment]
@aramase Can you pls rebase and resolve conflicts?

File: main.go [hunk 4]
```diff
@@ -16,44 +19,65 @@ var (
...
+
+	klog.Flush()
 	os.Exit(0)
 }
 
-func log(msg string) {
-	fmt.Fprintf(os.Stdout, msg)
+func setupLogFile() (*os.File, error) {
+	fileName := os.Getenv("PROVIDER_LOG_FILE")
+	if fileName == "" {
```
Comment (ritazh) [inline main.go]: since you set `PROVIDER_LOG_FILE` in `init()`, when will this be true?

Comment (aramase) [reply main.go]: My initial plan was to add it as an env var while building binary which is why I added the check. With `init` it should never be true.

Comment (ritazh) [reply main.go]: If this is never true, maybe we should remove this?

**Pull request merged**